From ca63a2b20de89be5d52092a24601e3c6bc79293c Mon Sep 17 00:00:00 2001
From: Rick Chen <rick@andestech.com>
Date: Wed, 30 Nov 2022 16:22:08 +0800
Subject: [PATCH 22/34] riscv: qilai: Support larger Image for Fast Boot

It can support larger than 32MB Kernel Image for Fast Boot.

Signed-off-by: Rick Chen <rick@andestech.com>
---
 configs/qilai_invecas_fastboot_defconfig | 4 ++--
 configs/qilai_rv64_fastboot_defconfig    | 4 ++--
 include/configs/andes45-qilai-invecas.h  | 4 ++--
 include/configs/andes45-qilai.h          | 4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/configs/qilai_invecas_fastboot_defconfig b/configs/qilai_invecas_fastboot_defconfig
index 833e0e6e42..2bb5c91d7a 100644
--- a/configs/qilai_invecas_fastboot_defconfig
+++ b/configs/qilai_invecas_fastboot_defconfig
@@ -4,7 +4,7 @@ CONFIG_SYS_MALLOC_LEN=0x800000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_ENV_SECT_SIZE=0x1000
 CONFIG_DEFAULT_DEVICE_TREE="qilai_64"
-CONFIG_SPL_SYS_MALLOC_F_LEN=0x1800000
+CONFIG_SPL_SYS_MALLOC_F_LEN=0x2800000
 CONFIG_SPL=y
 CONFIG_TARGET_ANDES45_QILAI_INVECAS=y
 CONFIG_ARCH_RV64I=y
@@ -14,7 +14,7 @@ CONFIG_SPL_LOAD_FIT_CONFIG="binman_d.dtsi"
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_SYS_LOAD_ADDR=0x400100000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x406000000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x410000000
 CONFIG_BOOTDELAY=3
 CONFIG_MISC_INIT_R=y
 CONFIG_SPL_BOARD_INIT=y
diff --git a/configs/qilai_rv64_fastboot_defconfig b/configs/qilai_rv64_fastboot_defconfig
index f023b69298..c6c5117a4a 100644
--- a/configs/qilai_rv64_fastboot_defconfig
+++ b/configs/qilai_rv64_fastboot_defconfig
@@ -4,7 +4,7 @@ CONFIG_SYS_MALLOC_LEN=0x800000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_ENV_SECT_SIZE=0x1000
 CONFIG_DEFAULT_DEVICE_TREE="qilai_64"
-CONFIG_SPL_SYS_MALLOC_F_LEN=0x1800000
+CONFIG_SPL_SYS_MALLOC_F_LEN=0x2800000
 CONFIG_SPL=y
 CONFIG_TARGET_ANDES45_QILAI=y
 CONFIG_ARCH_RV64I=y
@@ -14,7 +14,7 @@ CONFIG_SPL_LOAD_FIT_CONFIG="binman_d.dtsi"
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_SYS_LOAD_ADDR=0x400100000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x406000000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x410000000
 CONFIG_BOOTDELAY=3
 CONFIG_MISC_INIT_R=y
 CONFIG_SPL_BOARD_INIT=y
diff --git a/include/configs/andes45-qilai-invecas.h b/include/configs/andes45-qilai-invecas.h
index 703dc7d157..074aed6bac 100644
--- a/include/configs/andes45-qilai-invecas.h
+++ b/include/configs/andes45-qilai-invecas.h
@@ -11,7 +11,7 @@
 
 #ifdef CONFIG_SPL
 #define CONFIG_SPL_MAX_SIZE		0x00100000
-#define CONFIG_SPL_BSS_START_ADDR	0x404000000
+#define CONFIG_SPL_BSS_START_ADDR	0x400400000
 #define CONFIG_SPL_BSS_MAX_SIZE		0x00100000
 
 #ifdef CONFIG_SPL_MMC
@@ -78,7 +78,7 @@
 #define CONFIG_SYS_NS16550_CLK		50000000
 
 /* Init Stack Pointer */
-#define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x4000000 - \
+#define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x10000000 - \
 					GENERATED_GBL_DATA_SIZE)
 
 /* use CFI framework */
diff --git a/include/configs/andes45-qilai.h b/include/configs/andes45-qilai.h
index f5b58fc26c..f999531457 100644
--- a/include/configs/andes45-qilai.h
+++ b/include/configs/andes45-qilai.h
@@ -11,7 +11,7 @@
 
 #ifdef CONFIG_SPL
 #define CONFIG_SPL_MAX_SIZE		0x00100000
-#define CONFIG_SPL_BSS_START_ADDR	0x404000000
+#define CONFIG_SPL_BSS_START_ADDR	0x400400000
 #define CONFIG_SPL_BSS_MAX_SIZE		0x00100000
 
 #ifdef CONFIG_SPL_MMC
@@ -78,7 +78,7 @@
 #define CONFIG_SYS_NS16550_CLK		19660800
 
 /* Init Stack Pointer */
-#define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x4000000 - \
+#define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x10000000 - \
 					GENERATED_GBL_DATA_SIZE)
 
 /* use CFI framework */
-- 
2.34.1

